main.py

```diff
@@ -2,9 +2,53 @@
 import os
 import hashlib
 import sys
-load_env()
+from reportlab.pdfgen import canvas
 
 
+load_env(env_path=r'.\example.env')
+
+def is_text_file(file_path):
+    text_file_extensions = ['.txt', '.md', '.go', '.py', '.java', '.html', '.css', '.js', '.mod', '.sum']  # Add more as needed
+    return any(file_path.lower().endswith(ext) for ext in text_file_extensions)
+
+def combine_txt_files_and_create_pdf(source_directory, output_file, pdf_output, separator='**'):
+    separator_line = separator * 40 + '\n'
+    
+    # Initialize a list to store combined text
+    combined_text = []
+
+    with open(output_file, 'w', encoding='utf-8') as outfile:
+        for root, dirs, files in os.walk(source_directory):
+            for filename in files:
+                if filename.endswith('.txt'):
+                    file_path = os.path.join(root, filename)
+                    with open(file_path, 'r', encoding='utf-8') as infile:
+                        content = infile.read()
+                        combined_text.append(separator_line)
+                        combined_text.append(f"{filename.center(len(separator_line))}\n")
+                        combined_text.append(separator_line)
+                        combined_text.append(content + '\n')
```

Comment:
What if ```content``` already has a newline character? This will not work well with what you've written on line 41:

```python
for subline in line.split('\n')
```